eth_tester/main.py

```diff
@@ -64,6 +68,29 @@ def get_default_fork_blocks():
     }
 
 
+def handle_auto_mining(func):
+    @functools.wraps(func)
+    def func_wrapper(self, *args, **kwargs):
+        if not self.auto_mine_transactions:
+            snapshot = self.take_snapshot()
+
+        transaction_hash = func(self, *args, **kwargs)
+
+        if self.auto_mine_transactions:
+            self.mine_block()
+        else:
+            pending_transaction = self.get_transaction_by_hash(transaction_hash)
+            # Remove any pending transactions with the same nonce
+            self._pending_transactions = [tx for tx in self._pending_transactions
+                                          if not (pending_transaction['nonce'] == tx['nonce']
+                                                  and pending_transaction['from'] == tx['from'])]
```

Comment:
and chainId equals?

Reply:
Probably, but way less important than the others.

Reply:
I expect the node to reject a transaction if you send it with a different chainid than the node is syncing with, so the handling of failed transactions should be sufficient.